Source/missiles.cpp

```diff
@@ -3148,24 +3157,28 @@ void MI_Dummy(int i)
 void MI_Golem(int mi)
 {
 	int src = Missiles[mi]._misource;
-	if (Monsters[src].position.tile.x == 1 && Monsters[src].position.tile.y == 0) {
+	if (Monsters[src].position.tile == Point { 1, 0 }) {
 		for (int i = 0; i < 6; i++) {
 			int k = CrawlNum[i];
 			int ck = k + 2;
 			for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
 				const int8_t *ct = &CrawlTable[ck];
 				int tx = Missiles[mi]._miVar4 + *(ct - 1);
 				int ty = Missiles[mi]._miVar5 + *ct;
-				if (0 < tx && tx < MAXDUNX && 0 < ty && ty < MAXDUNY) {
-					int dp = dPiece[tx][ty];
-					if (LineClearMissile({ Missiles[mi]._miVar1, Missiles[mi]._miVar2 }, { tx, ty })) {
-						if (dMonster[tx][ty] == 0 && !nSolidTable[dp] && dObject[tx][ty] == 0) {
-							i = 6;
-							SpawnGolum(src, { tx, ty }, mi);
-							break;
-						}
-					}
-				}
+
+				if (!InDungeonBounds({ tx, ty }))
+					continue;
+
+				int dp = dPiece[tx][ty];

```

Comment:
As far as I can tell, `LineClearMissile` doesn't modify `dPiece` (or `tx, ty`). This declaration can move to above the conditional on line 3176.

Reply:
Made it `nSolidTable[dPiece[tx][ty]]` like in most other use cases.